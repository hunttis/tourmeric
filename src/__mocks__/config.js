export const config = {
  default: {
    titleText: 'Foo',
    apiKey: 'Foo',
    authDomain: 'Foo',
    databaseURL: 'Foo',
    projectId: 'Foo',
    storageBucket: 'Foo',
    messagingSenderId: 'Foo',
    fileMetadataFactory: (uploadRes) => {
      const { metadata: { name, fullPath, downloadURLs } } = uploadRes;
      return {
        name,
        fullPath,
        downloadURL: downloadURLs,
      };
    },
  },
};

exports.get = function get() {
  return config.default;
};
