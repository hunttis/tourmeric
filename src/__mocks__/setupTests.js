import { configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16'; // eslint-disable-line

configure({ adapter: new Adapter() });
