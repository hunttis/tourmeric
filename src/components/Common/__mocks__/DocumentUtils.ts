
export default function scrollIntoview(_elementId: string) {
  console.log('scrolling');
}

export function setListener(_eventType: string, _callback: () => {}) {
  console.log('adding listener');
}

export function removeClassFromHtml(className: string) {
  console.log('removing ', className);
}

export function addClassToHtml(className: string) {
  console.log('adding ', className);
}