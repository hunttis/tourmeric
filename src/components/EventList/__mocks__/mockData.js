export const mockEventId = 'mockEventId1';

export const mockUserId = 'mockUserId';

export const mockProfile = {
  avatarUrl: 'https://foo.com/picture',
  dciNumber: '123512351112',
  displayName: 'Test Person',
  email: 'fake@fake.com',
  firstName: 'Test',
  lastName: 'Person',
  providerData: [{ displayName: 'Test Person',
    email: 'fake@fake.com',
    photoURL: 'https://foo.com/picture',
    providerId: 'facebook.com',
    uid: '1234567890' }],
  role: 'admin',
  isEmpty: false,
  isLoaded: true,
};

export const mockSingleEvent = {
  category: 'category1',
  createDate: '2018-05-24T17:00:46.500Z',
  date: '2018-09-24',
  entryFee: '10',
  format: 'Miniatures',
  link: 'http://www.google.fi',
  name: 'Warhammer thing',
  notes: 'Bring your war face!',
  playerSlots: '10',
  prizes: 'Glory!',
  published: true,
  rulesLevel: 'Casual',
  time: 'Kill everything',
};

export const mockUnorderedEvents = {
  mockEvent2: {
    category: 'category2',
    createDate: '2018-05-20T16:46:32.009Z',
    date: '2018-05-20',
    entryFee: '6',
    format: 'Standard',
    name: 'Tapahtuma',
    notes: 'FIJoijgasdsdgasdggagads',
    playerSlots: '14',
    prizes: 'Foooooooooooooooo',
    rulesLevel: 'Competitive',
    time: '10:00',
  },
  mockEvent3: {
    category: 'category2',
    createDate: '2018-05-20T16:04:29.595Z',
    date: '2018-05-21',
    entryFee: '5',
    format: 'Modern',
    link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
    name: 'Monday Modern',
    notes: 'Bring a friend!',
    playerSlots: '31',
    prizes: 'Three (3) OMG credits per player into the prizepool.',
    published: true,
    rulesLevel: 'Regular',
    time: '19:00',
  },
};

export const mockEvents = [
  { key: 'mockEvent2',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:46:32.009Z',
      date: '2018-05-20',
      entryFee: '6',
      format: 'Standard',
      name: 'Tapahtuma',
      notes: 'FIJoijgasdsdgasdggagads',
      playerSlots: '14',
      prizes: 'Foooooooooooooooo',
      rulesLevel: 'Competitive',
      time: '10:00',
    } },
  { key: 'mockEvent3',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:04:29.595Z',
      date: '2018-05-21',
      entryFee: '5',
      format: 'Modern',
      link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
      name: 'Monday Modern',
      notes: 'Bring a friend!',
      playerSlots: '31',
      prizes: 'Three (3) OMG credits per player into the prizepool.',
      published: true,
      rulesLevel: 'Regular',
      time: '19:00',
    } },
  { key: 'mockEvent4',
    value: {
      category: '-LDIw-1LFgiM0EJZGfus',
      createDate: '2018-05-24T20:58:06.938Z',
      date: '2018-05-24',
      entryFee: '5',
      format: 'League',
      link: 'jokulinkki.fi',
      name: 'Necromunda League',
      notes: 'Shame! Shame! Shame! *DING DING DING*',
      playerSlots: '12',
      prizes: 'I am Loot.',
      published: true,
      rulesLevel: 'Casual',
      time: '17:00',
    } },
  { key: 'mockEvent5',
    value: {
      category: '-LAlvCmIWNzYKpa7-7HI',
      createDate: '2018-05-24T17:01:53.909Z',
      date: '2018-05-26',
      entryFee: '0',
      format: 'Board game',
      link: 'https://geekandsundry.com/international-tabletop-day-april-28-2018/',
      name: 'Lautapeli Lauantai',
      notes: 'Bring a friend and get a surprise gift!',
      playerSlots: '30',
      prizes: "A Supi's surprise prize for all participants!",
      published: true,
      rulesLevel: 'Casual',
      time: '00:00',
    } },
  { key: 'mockEventId1',
    value: {
      category: 'category1',
      createDate: '2018-05-24T17:00:46.500Z',
      date: '2018-09-24',
      entryFee: '10',
      format: 'Miniatures',
      link: 'http://www.google.fi',
      name: 'Warhammer thing',
      notes: 'Bring your war face!',
      playerSlots: '10',
      prizes: 'Glory!',
      published: true,
      rulesLevel: 'Casual',
      time: 'Kill everything',
    } },
];

export const mockSettings = {
  activeLogo: 'http://foo.com/image.png',
  dateFormat: 'DD.MM.YYYY',
  pageSubtitle: '',
  pageTitle: '',
  subtitleTextColor: '#000',
  theme: 'darkly',
  titleBarAngle: '150',
  titleBarColor: '#000',
  titleBarColor2: '#ec1c22',
  titleBarPercentage: '50',
  titleTextColor: '',
};

export const mockUploadedLogos =
  {
    15271953483568546: { downloadURL: 'http://foo.com/image.png' },
    '152719372155055672': { downloadURL: 'http://foo.com/image.png' },
    '152719272536668394': { downloadURL: 'http://foo.com/image.png' },
    '1527227370219137123': { downloadURL: 'http://foo.com/image.png' },
    '152722965771313789': { downloadURL: 'http://foo.com/image.png' },
    '152722747904159438': { downloadURL: 'http://foo.com/image.png' },
  };

export const mockParticipations = { mockEventId1: { mockUserId: { comment: 'Jee',
  date: '2018-06-23T09:40:26.992Z',
  firstName: 'Test',
  lastName: 'Person',
  userId: 'mockUserId' } } };

export const mockCategories = { category1: { abbreviation: 'WH40K',
  logo: '1527227370219137123',
  name: 'Warhammer 40,000',
  type: 'Miniatures' },
category2: { abbreviation: 'MtG',
  logo: '152722965771313789',
  name: 'Magic: The Gathering',
  type: 'TCG' },
'-LAlvCmIWNzYKpa7-7HI': { abbreviation: 'Lautapelit',
  logo: '152719272536668394',
  name: 'Lautapelit / Board games' },
'-LCmESny8m9JVEFQ_qOt': { abbreviation: 'PKM',
  logo: '152722747904159438',
  name: 'Pok√©mon',
  type: 'TCG' },
'-LDImTAGxkzrorL0tQdi': { abbreviation: 'VG',
  logo: '152719372155055672',
  name: 'Videogames' },
'-LDIw-1LFgiM0EJZGfus': { abbreviation: 'Necromunda',
  logo: '15271953483568546',
  name: 'Necromunda',
  type: 'Miniatures' } };
