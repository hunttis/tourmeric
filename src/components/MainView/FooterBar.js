import React, { Component, Fragment } from 'react';
import { Translate } from 'react-localize-redux';
import _ from 'lodash';
import PropTypes from 'prop-types';

import PrivacyPolicy from './PrivacyPolicy/PrivacyPolicy-container';

export default class FooterBar extends Component {

  foo() {}

  render() {

    const { settings } = this.props;

    const showingSponsors = _.get(settings, 'showSponsors', false);
    const hasAtLeastOneSponsor = showingSponsors && !_.isEmpty(_.get(settings, 'footer', {}));

    return (
      <footer className="footer less-bottompadding">
        <div className="content">
          <div className="columns">

            {hasAtLeastOneSponsor &&
            <Fragment>
              <div className="column has-text-centered is-hidden-desktop">
                <Translate id="sponsoredby" />:
              </div>
              <div className="column has-text-left is-hidden-mobile">
                <Translate id="sponsoredby" />:
              </div>
            </Fragment>
            }
            {(showingSponsors && _.get(settings, 'footer.first.image')) &&
            <div className="column is-vcentered">
              <figure className="image is-paddingless is-marginless">
                <a href={settings.footer.first.link} target="_blank" rel="noopener noreferrer">
                  <img className="footerImage" src={settings.footer.first.image} alt="" />
                </a>
              </figure>
            </div>
            }
            {(showingSponsors && _.get(settings, 'footer.second.image')) &&
            <div className="column">
              <figure className="image is-paddingless is-marginless">
                <a href={settings.footer.second.link} target="_blank" rel="noopener noreferrer">
                  <img className="footerImage" src={settings.footer.second.image} alt="" />
                </a>
              </figure>
            </div>
            }
            {(showingSponsors && _.get(settings, 'footer.third.image')) &&
            <div className="column">
              <figure className="image is-paddingless is-marginless">
                <a href={settings.footer.third.link} target="_blank" rel="noopener noreferrer">
                  <img className="footerImage" src={settings.footer.third.image} alt="" />
                </a>
              </figure>
            </div>
            }
            <div className="column has-text-centered is-hidden-desktop">
              <PrivacyPolicy showAcceptance={false} />
            </div>
            <div className="column has-text-right is-hidden-mobile">
              <PrivacyPolicy showAcceptance={false} />
            </div>
          </div>
        </div>
      </footer>
    );
  }
}

FooterBar.propTypes = {
  settings: PropTypes.object,
};
