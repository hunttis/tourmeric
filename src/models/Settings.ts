
export interface Settings {
  activeLocationImage?: string;
  activeLogo: string;
  browserTitle?: string;
  companyinfo?: string;
  dateFormat: string;
  features?: Features;
  introText?: string;
  location?: Location;
  openingHours?: OpeningHours;
  pageSubtitle: string;
  pageTitle: string;
  privacyPolicy?: string;
  showSponsors?: boolean;
  subtitleTextColor: string;
  theme: string;
  titleBarAngle: string;
  titleBarColor: string;
  titleBarColor2: string;
  titleBarPercentage: string;
  titleTextColor: string;
  footer?: {[key: string]: Footer};
  locationImage?: string;
}

interface Features {
  events: Feature;
  highlights: Feature;
  storeinfo: Feature;
}

interface Feature {
  active: boolean;
  introtext: boolean;
}

interface Location {
  address: string;
  directions: string;
  email: string;
  phone: string;
}


export interface OpeningHours {
  additionalinfo: string;
  monday: string;
  tuesday: string;
  wednesday: string;
  thursday: string;
  friday: string;
  saturday: string;
  sunday: string;
}

export interface Footer {
  link: string;
  image: string;
  text: string;
}
