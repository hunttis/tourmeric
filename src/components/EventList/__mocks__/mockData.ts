import { FirebaseProfile, Participation, User, ProviderData, Location } from '~/models/ReduxState';
import { TourmericEvent } from '~/models/Events';
import { Settings } from '~/models/Settings';
import { UploadedFile, Category } from '~/models/Category';

export const mockEventId = 'mockEvent1';

export const mockUserId = 'mockUserId';

export const mockLocation: Location = {
  pathname: '/events/2018/10',
  search: 'search',
  hash: 'hash',
  key: 'key',
};

export const mockProviderData: ProviderData = {
  displayName: 'MockName',
  email: 'mock@userton.com',
  photoURL: 'http://url',
  providerId: 'facebook',
  uid: '1234567890',
};

export const mockUser: User = {
  avatarUrl: 'http://www.google.fi/google.png',
  displayName: 'Mock Userton',
  firstName: 'Mock',
  lastName: 'Userton',
  providerData: mockProviderData,
  email: 'mock@userton.com',
  username: 'Mockie',
  dciNumber: '123456',
  role: 'user',
  active: true,
};

export const mockProfile: FirebaseProfile = {
  avatarUrl: 'https://foo.com/picture',
  dciNumber: '123512351112',
  displayName: 'Test Person',
  email: 'fake@fake.com',
  firstName: 'Test',
  lastName: 'Person',
  providerData: [{
    displayName: 'Test Person',
    email: 'fake@fake.com',
    photoURL: 'https://foo.com/picture',
    providerId: 'facebook.com',
    uid: '1234567890',
  }],
  role: 'admin',
  isEmpty: false,
  isLoaded: true,
};

export const mockSingleEvent: TourmericEvent = {
  category: 'category1',
  createDate: '2018-05-24T17:00:46.500Z',
  date: '2018-09-24',
  entryFee: '10',
  format: 'Miniatures',
  link: 'http://www.google.fi',
  name: 'Warhammer thing',
  notes: 'Bring your war face!',
  playerSlots: '10',
  prizes: 'Glory!',
  published: true,
  rulesLevel: 'Casual',
  time: 'Kill everything',
  eventType: 'single',
};

export const mockUnorderedEvents: {[key: string]: TourmericEvent} = {
  mockEvent1: mockSingleEvent,
  mockEvent2: {
    category: 'category2',
    createDate: '2018-05-20T16:46:32.009Z',
    date: '2018-05-20',
    entryFee: '6',
    format: 'Standard',
    name: 'Tapahtuma',
    notes: 'FIJoijgasdsdgasdggagads',
    playerSlots: '14',
    prizes: 'Foooooooooooooooo',
    rulesLevel: 'Competitive',
    time: '10:00',
  },
  mockEvent3: {
    category: 'category2',
    createDate: '2018-05-20T16:04:29.595Z',
    date: '2018-05-21',
    entryFee: '5',
    format: 'Modern',
    link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
    name: 'Monday Modern',
    notes: 'Bring a friend!',
    playerSlots: '31',
    prizes: 'Three (3) OMG credits per player into the prizepool.',
    published: true,
    rulesLevel: 'Regular',
    time: '19:00',
  },
};

export const mockEvents: {key: string, value: TourmericEvent}[] = [
  { key: 'mockEvent1',
    value: mockSingleEvent },
  { key: 'mockEvent2',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:46:32.009Z',
      date: '2018-05-20',
      entryFee: '6',
      format: 'Standard',
      name: 'Tapahtuma',
      notes: 'FIJoijgasdsdgasdggagads',
      playerSlots: '14',
      prizes: 'Foooooooooooooooo',
      rulesLevel: 'Competitive',
      time: '10:00',
    } },
  { key: 'mockEvent3',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:04:29.595Z',
      date: '2018-05-21',
      entryFee: '5',
      format: 'Modern',
      link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
      name: 'Monday Modern',
      notes: 'Bring a friend!',
      playerSlots: '31',
      prizes: 'Three (3) OMG credits per player into the prizepool.',
      published: true,
      rulesLevel: 'Regular',
      time: '19:00',
    } },
  { key: 'mockEvent4',
    value: {
      category: '-LDIw-1LFgiM0EJZGfus',
      createDate: '2018-05-24T20:58:06.938Z',
      date: '2018-05-24',
      entryFee: '5',
      format: 'League',
      link: 'jokulinkki.fi',
      name: 'Necromunda League',
      notes: 'Shame! Shame! Shame! *DING DING DING*',
      playerSlots: '12',
      prizes: 'I am Loot.',
      published: true,
      rulesLevel: 'Casual',
      time: '17:00',
    } },
  { key: 'mockEvent5',
    value: {
      category: '-LAlvCmIWNzYKpa7-7HI',
      createDate: '2018-05-24T17:01:53.909Z',
      date: '2018-05-26',
      entryFee: '0',
      format: 'Board game',
      link: 'https://geekandsundry.com/international-tabletop-day-april-28-2018/',
      name: 'Lautapeli Lauantai',
      notes: 'Bring a friend and get a surprise gift!',
      playerSlots: '30',
      prizes: "A Supi's surprise prize for all participants!",
      published: true,
      rulesLevel: 'Casual',
      time: '00:00',
    } },
  { key: 'mockEventId1',
    value: {
      category: 'category1',
      createDate: '2018-05-24T17:00:46.500Z',
      date: '2018-09-24',
      entryFee: '10',
      format: 'Miniatures',
      link: 'http://www.google.fi',
      name: 'Warhammer thing',
      notes: 'Bring your war face!',
      playerSlots: '10',
      prizes: 'Glory!',
      published: true,
      rulesLevel: 'Casual',
      time: 'Kill everything',
    } },
];

export const mockSettings: Settings = {
  activeLogo: 'http://foo.com/image.png',
  dateFormat: 'DD.MM.YYYY',
  pageSubtitle: '',
  pageTitle: '',
  subtitleTextColor: '#000',
  theme: 'darkly',
  titleBarAngle: '150',
  titleBarColor: '#000',
  titleBarColor2: '#ec1c22',
  titleBarPercentage: '50',
  titleTextColor: '',
};

export const mockUploadedLogos: {[key: string]: UploadedFile} =
  {
    15271953483568546: { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152719372155055672': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152719272536668394': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '1527227370219137123': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152722965771313789': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152722747904159438': { downloadURL: 'http://foo.com/image.png', name: 'image' },
  };

export const mockParticipations: {[key: string]: Participation} = {
  mockEventId1: {
    mockUserId: {
      comment: 'Jee',
      date: '2018-06-23T09:40:26.992Z',
      firstName: 'Test',
      lastName: 'Person',
      userId: 'mockUserId',
    },
  },
};

export const mockCategories: {[key: string]: Category} = {
  category1: {
    abbreviation: 'WH40K',
    logo: '1527227370219137123',
    name: 'Warhammer 40,000',
    type: 'Miniatures',
  },
  category2: {
    abbreviation: 'MtG',
    logo: '152722965771313789',
    name: 'Magic: The Gathering',
    type: 'TCG',
  },
  '-LAlvCmIWNzYKpa7-7HI': {
    abbreviation: 'Lautapelit',
    logo: '152719272536668394',
    name: 'Lautapelit / Board games',
    type: 'Boardgame',
  },
  '-LCmESny8m9JVEFQ_qOt': {
    abbreviation: 'PKM',
    logo: '152722747904159438',
    name: 'Pok√©mon',
    type: 'TCG',
  },
  '-LDImTAGxkzrorL0tQdi': {
    abbreviation: 'VG',
    logo: '152719372155055672',
    name: 'Videogames',
    type: 'Videogames',
  },
  '-LDIw-1LFgiM0EJZGfus': {
    abbreviation: 'Necromunda',
    logo: '15271953483568546',
    name: 'Necromunda',
    type: 'Miniatures',
  },
};
