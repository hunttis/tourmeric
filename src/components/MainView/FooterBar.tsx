import React from 'react';
import { FormattedMessage, IntlShape } from "react-intl";
import _ from 'lodash';
import { isLoaded } from 'react-redux-firebase';
import PrivacyPolicy from './PrivacyPolicy/PrivacyPolicy-container';
import { Settings } from '../../models/Settings';

interface Props {
  settings: Settings;
}

const FooterBar = ({ settings }: Props) => {

  const showingSponsors = _.get(settings, 'showSponsors', false);
  const hasAtLeastOneSponsor = showingSponsors && !_.isEmpty(_.get(settings, 'footer', {}));

  if (!isLoaded(settings)) {
    return <div />;
  }

  return (
    <footer className="footer less-bottompadding">
      <div className="content">
        <div className="columns is-multiline">

          {hasAtLeastOneSponsor &&
            <>
              <div className="column is-12 has-text-centered is-hidden-tablet">
                <FormattedMessage id="sponsoredby" />:
              </div>
              <div className="column is-6 has-text-left is-hidden-mobile">
                <FormattedMessage id="sponsoredby" />:
              </div>
              <div className="column is-6 has-text-right is-hidden-mobile">
                <PrivacyPolicy showAcceptance={false} />
              </div>
            </>
          }

          {showingSponsors &&
            <>
              {settings.footer && Object.entries(settings.footer).map((footerEntry) => {

                const key = footerEntry[0];
                const value = footerEntry[1];

                return (
                  <div key={`Footer-${key}`} className="column sponsor">
                    <figure className="image is-paddingless is-marginless">
                      <a href={value.link} target="_blank" rel="noopener noreferrer" title={value.text}>
                        <img className="footerImage" src={value.image} alt="" />
                      </a>
                    </figure>
                  </div>
                );
              })}
            </>
          }

          <div className="column is-12 has-text-centered is-hidden-tablet">
            <PrivacyPolicy showAcceptance={false} />
          </div>

        </div>
      </div>
    </footer>
  );
};

export default FooterBar;
