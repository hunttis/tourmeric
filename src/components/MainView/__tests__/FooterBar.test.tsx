/* eslint-disable import/first */

import React from 'react';
import { shallow } from 'enzyme';
import rrf from 'react-redux-firebase';
import FooterBar from '../FooterBar';
import { Settings } from '~/models/Settings';
import { emptyMockSettings as settings } from '~/components/EventList/__mocks__/mockData';

jest.mock('react-redux-firebase');

describe('FooterBar tests', () => {
  beforeEach(() => {
    rrf.isLoaded = jest.fn(() => true);
  });

  it('renders when Settings are not loaded', () => {
    rrf.isLoaded = jest.fn(() => false);

    const testSettings = {
      showingSponsors: false,
      ...settings,
    };
    const footerBar = shallow(<FooterBar settings={testSettings} />);

    expect(footerBar.find('.sponsor').length).toBe(0);
    expect(footerBar.find('.column').length).toBe(0);
    expect(footerBar).toMatchInlineSnapshot('<div />');
  });

  it('Snapshot, sponsors not shown', () => {

    const testSettings = {
      showingSponsors: false,
      ...settings,
    };
    const footerBar = shallow(<FooterBar settings={testSettings} />);

    expect(footerBar.find('.sponsor').length).toBe(0);
    expect(footerBar.find('.column').length).toBe(1);
    expect(footerBar).toMatchSnapshot();
  });

  it('Snapshot, sponsors shown, one sponsor', () => {
    const testSettings: Settings = {
      footer: {
        first: { image: 'foo.png', link: 'http://link', text: 'first' },
      },
      showSponsors: true,
      ...settings,
    };
    const footerBar = shallow(<FooterBar settings={testSettings} />);
    expect(footerBar.find('.sponsor').length).toBe(1);
    expect(footerBar.find('.column').length).toBe(5);
  });

  it('Snapshot, sponsors shown, two sponsors', () => {
    const testSettings: Settings = {
      footer: {
        first: { image: 'foo.png', link: 'http://link', text: 'first' },
        second: { image: 'foo2.png', link: 'http://link2', text: 'second' },
      },
      showSponsors: true,
      ...settings,
    };
    const footerBar = shallow(<FooterBar settings={testSettings} />);
    expect(footerBar.find('.sponsor').length).toBe(2);
    expect(footerBar.find('.column').length).toBe(6);
  });

  it('Snapshot, sponsors shown, three sponsors', () => {
    const testSettings: Settings = {
      footer: {
        first: { image: 'foo.png', link: 'http://link', text: 'first' },
        second: { image: 'foo2.png', link: 'http://link2', text: 'second' },
        third: { image: 'foo3.png', link: 'http://link3', text: 'third' },
      },
      showSponsors: true,
      ...settings,
    };
    const footerBar = shallow(<FooterBar settings={testSettings} />);
    expect(footerBar.find('.sponsor').length).toBe(3);
    expect(footerBar.find('.column').length).toBe(7);
  });
});
