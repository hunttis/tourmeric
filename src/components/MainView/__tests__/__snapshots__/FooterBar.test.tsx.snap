// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterBar tests Snapshot, sponsors not shown 1`] = `
<footer
  className="footer less-bottompadding"
>
  <div
    className="content"
  >
    <div
      className="columns is-multiline"
    >
      <div
        className="column is-12 has-text-centered is-hidden-tablet"
      >
        <Connect(Connect(PrivacyPolicy))
          showAcceptance={false}
        />
      </div>
    </div>
  </div>
</footer>
`;
